### PR TITLE
feat(integrations): GitLab limit issue search to project scope and add issue search by iid.

### PR DESCRIPTION
**Pull request by lauryndbrown**

### CODE REVIEW

File: src/sentry/integrations/gitlab/issues.py [hunk 1]
```diff
@@ -35,13 +44,9 @@ def create_default_repo_choice(self, default_repo):
         return (project['id'], project['name_with_namespace'])
 
     def get_create_issue_config(self, group, **kwargs):
-        params = kwargs.get('params', {})
-        defaults = self.get_project_defaults(group.project_id)
-        kwargs['repo'] = params.get('project', defaults.get('project'))
```
Comment (markstory) [inline src/sentry/integrations/gitlab/issues.py]: Is the `repo` key needed in `get_create_issue_config()`?

File: src/sentry/integrations/gitlab/issues.py [hunk 2]
```diff
@@ -92,20 +97,31 @@ def create_issue(self, data, **kwargs):
...
+
         org = group.organization
         autocomplete_url = reverse(
             'sentry-extensions-gitlab-search', args=[org.slug, self.model.id],
         )
 
         return [
+            {
+                'name': 'project',
+                'label': 'GitLub Project',
```
Comment (markstory) [inline src/sentry/integrations/gitlab/issues.py]:
```suggestion
                'label': 'GitLab Project',
```

**Review by markstory — COMMENTED** [review]
It would be good to have tests for the new endpoint behaviors so we have less chance of regressing in the future.

**Comment by lauryndbrown** [comment]
https://cl.ly/a81f4521304c

File: src/sentry/integrations/gitlab/issues.py (hunk 1 above)
Comment (lauryndbrown) [reply src/sentry/integrations/gitlab/issues.py]: Yes. So this was what I did to make it work with the default's code here: https://github.com/getsentry/sentry/blob/master/src/sentry/integrations/issues.py#L198:L226

Comment (markstory) [reply src/sentry/integrations/gitlab/issues.py]: Should the code be re-ordered then? The `repo` key isn't set until after `get_create_issue_config()` is called now.

**Comment by lauryndbrown** [comment]
@markstory let me know what you think about the tests now.

File: tests/sentry/integrations/gitlab/test_search.py [hunk 3]
```diff
@@ -80,14 +104,34 @@ def test_finds_project_results(self):
...
+            responses.GET,
+            'https://example.gitlab.com/api/v4/projects/5/issues?scope=all&search=11',
+            json=[]
+        )
+        resp = self.client.get(
+            self.url,
+            data={
+                'field': 'externalIssue',
+                'query': '11',
+                'project': '5',
```
Comment (markstory) [inline tests/sentry/integrations/gitlab/test_search.py]: You might want one more test case for the scenario when user manages to send a request off with no project.

**Review by markstory — APPROVED** [review]

**Pull request merged**